src/assets/javascripts/components/content/code/mermaid/index.ts

```diff
@@ -69,7 +69,7 @@ let sequence = 0
  */
 function fetchScripts(): Observable<void> {
   return typeof mermaid === "undefined" || mermaid instanceof Element
-    ? watchScript("https://unpkg.com/mermaid@9.1.7/dist/mermaid.min.js")
+    ? watchScript("_static/mermaid.min.js") // sphinx-immaterial uses cached dist
```

Comment:
This path needs to take into account that the document may be a in sub-dir.

Reply:
I was under the impression that the path specified here is relative to the base site path. The python code (in \_\_init\_\_.py) doesn't override user-defined mathjax path, so I'm not sure how to get this JS to coincide with the Py side without overriding the user-defined path (which will ignore any custom version of mathjax dist that users try to employ).

Reply:
I believe the mathjax path is indeed handled by sphinx automatically, but this code is for mermaid and is using the mkdocs-material `watchScript` function which just seems to copy the argument directly as the `<script src="...">` attribute.

Reply:
> but this code is for mermaid 

oops! 😆 

> using the mkdocs-material `watchScript` function which just seems to copy the argument directly as the `<script src="...">` attribute.

Ah yes. Currently, there's no way to specify where the mermaid dist is being copied from (by Sphinx at docs' build time). The build script should always copy the mermaid dist to the same folder for theme dist. So, I think the subfolder concern is null, yes?

Reply:
The issue is if we have foo/bar.rst

Reply:
Oh, I see. Good catch! I'm not sure how I would go about specifying the full path to the site base. I'll have to investigate further...

Reply:
Any suggestion as to how to do this for both local `file::///` and web-hosted `https://`? I see `window.location.origin` only works for the later.

Reply:
You can use the approach from sphinx_search.ts: https://github.com/jbms/sphinx-immaterial/blob/d91eacc94250c7fafb8b930ac744d3bbcffe50da/src/assets/javascripts/sphinx_search.ts#L30

Reply:
TBH, I don't fully understand how that works better than my previous attempts involving `location.href`, (must be something to do with the `URL` object), but I pushed the changes after verifying that they work in rst-cheatsheet/rst-cheatsheet.rst.

Reply:
Basically we have:

```
config.base = `${new URL(base_url, new URL(location.href))}`
```

where `base_url` is defined in Python as:

```
    num_slashes = builder.get_target_uri(pagename).count("/")
    if num_slashes == 0:
        base_url = "."
    else:
        base_url = "/".join(".." for _ in range(num_slashes))
```

The key is that we need an external source to tell us know many levels up we need to go.

Reply:
Ah, the `base_url` bit is what I was missing. Thanks for the extra details!